TiKV fail to compile on MacOS

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
master

### What operating system and CPU are you using?
MacOS, M1

### Steps to reproduce
make dev

### What did you expect?
Compile success

### What did happened?
```
  /.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/651a5c0/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:100:14: error: variable 'file_size' set but not used [-Werror,-Wunused-but-set-variable]
      uint64_t file_size = 0;
               ^
  1 error generated.
  make[5]: *** [CMakeFiles/titan.dir/src/db_impl.cc.o] Error 1
  make[5]: *** Waiting for unfinished jobs....
  make[4]: *** [CMakeFiles/titan.dir/all] Error 2
  make[3]: *** [CMakeFiles/titan.dir/rule] Error 2
  make[2]: *** [titan] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

```
% gcc --version
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

```
  /workspace/tirocks/tirocks-sys/rocksdb/options/db_options.h:90:12: error: definition of implicit copy assignment operator for 'MutableDBOptions' is deprecated because it has a user-declared copy constructor [-Werror,-Wdeprecated-copy]
    explicit MutableDBOptions(const MutableDBOptions& options) = default;
             ^
  /workspace/tirocks/tirocks-sys/rocksdb/db/db_impl/db_impl.cc:1082:27: note: in implicit copy assignment operator for 'rocksdb::MutableDBOptions' first required here
        mutable_db_options_ = new_options;
                            ^
  /workspace/tirocks/tirocks-sys/rocksdb/db/db_impl/db_impl_write.cc:112:12: error: variable 'valid_batches' set but not used [-Werror,-Wunused-but-set-variable]
      size_t valid_batches = 0;
             ^

```

--- a/db/compaction/compaction_picker_test.cc
+++ b/db/compaction/compaction_picker_test.cc
@@ -516,13 +516,11 @@ TEST_F(CompactionPickerTest, NeedsCompactionFIFO) {
 
   // verify whether compaction is needed based on the current
   // size of L0 files.
-  uint64_t current_size = 0;
   for (int i = 1; i <= kFileCount; ++i) {
     NewVersionStorage(1, kCompactionStyleFIFO);
     Add(0, i, ToString((i + 100) * 1000).c_str(),
-        ToString((i + 100) * 1000 + 999).c_str(),
-        kFileSize, 0, i * 100, i * 100 + 99);
-    current_size += kFileSize;
+        ToString((i + 100) * 1000 + 999).c_str(), kFileSize, 0, i * 100,
+        i * 100 + 99);
     UpdateVersionStorageInfo();
     ASSERT_EQ(fifo_compaction_picker.NeedsCompaction(vstorage_.get()),
               vstorage_->CompactionScore(0) >= 1);

--- a/db/db_test2.cc
+++ b/db/db_test2.cc
@@ -2199,7 +2199,6 @@ TEST_F(DBTest2, ReadAmpBitmapLiveInCacheAfterDBClose) {
     Close();
     Reopen(options);
 
-    uint64_t total_useful_bytes = 0;
     std::set<int> read_keys;
     std::string value;
     // Iter1: Read half the DB, Read even keys
@@ -2210,8 +2209,6 @@ TEST_F(DBTest2, ReadAmpBitmapLiveInCacheAfterDBClose) {
 
       if (read_keys.find(i) == read_keys.end()) {
         auto internal_key = InternalKey(key, 0, ValueType::kTypeValue);
-        total_useful_bytes +=
-            GetEncodedEntrySize(internal_key.size(), value.size());
         read_keys.insert(i);
       }
     }
@@ -2237,8 +2234,6 @@ TEST_F(DBTest2, ReadAmpBitmapLiveInCacheAfterDBClose) {
 
       if (read_keys.find(i) == read_keys.end()) {
         auto internal_key = InternalKey(key, 0, ValueType::kTypeValue);
-        total_useful_bytes +=
-            GetEncodedEntrySize(internal_key.size(), value.size());
         read_keys.insert(i);
       }
     }

--- a/options/db_options.h
+++ b/options/db_options.h
@@ -87,7 +87,6 @@ struct ImmutableDBOptions {
 
 struct MutableDBOptions {
   MutableDBOptions();
-  explicit MutableDBOptions(const MutableDBOptions& options) = default;
   explicit MutableDBOptions(const DBOptions& options);
 
   void Dump(Logger* log) const;

--- a/utilities/env_mirror.cc
+++ b/utilities/env_mirror.cc
@@ -27,13 +27,17 @@ class SequentialFileMirror : public SequentialFile {
     if (as == Status::OK()) {
       char* bscratch = new char[n];
       Slice bslice;
+#ifndef NDEBUG
       size_t off = 0;
+#endif
       size_t left = aslice.size();
       while (left) {
         Status bs = b_->Read(left, &bslice, bscratch);
+#ifndef NDEBUG
         assert(as == bs);
         assert(memcmp(bscratch, scratch + off, bslice.size()) == 0);
         off += bslice.size();
+#endif
         left -= bslice.size();
       }
       delete[] bscratch;

